zoo/birealnet.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2021 Loro Francesco
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__author__ = "Francesco Loro"
+__email__ = "francesco.official@gmail.com"
+__supervisor__ = "Danilo Pau"
+__email__ = "danilo.pau@st.com"
+
+# Download pretrained weight from:
+# Birealnet -> https://drive.google.com/file/d/1BuDYhydNAy-sFdvoh24gJbpA0zRDvGVq/view?usp=sharing
+
+import qkeras as q
+import tensorflow as tf
+import larq as lq
+from utils import compare_network, create_random_dataset, dump_network_to_json
+
+# Define path to the pre-trained weights
+PATH_BIREALNET = "weights/birealnet_weights.h5"
+BIREALNET_NAME = "biRealNet"
+
+
+class BirealNet:
```

Comment:
is it from this paper https://arxiv.org/pdf/1808.00278.pdf

Reply:
Yes, it is

Reply:
Yes, it is